src/BccCode.PubSub.Api/Controllers/SubscriptionsController.cs

```
@@ -71,6 +65,20 @@ public async Task<ActionResult<SubscribeResponseData>> Subscribe([FromBody] Subs
 
             await _repository.Create(resultSubscription);
```

I think we should handle the exception that could be thrown here if the subscription already exists. Otherwise it'll still just return a 500 on conflict.

---

No, the error thrown here is not handled.

---

"An exception thrown in a catch block will behave the same as an exception thrown without it"

---

Hmm, but I don't think you would want to throw the error again, I think we want to check what the error is and if it was a primary key conflict we woudl want to do something like
```cs
return new StatusCodeResult(StatusCodes.Status409Conflict);
```

---

It won't be a primary key conflict, because the database operation is atomic, and done first. I reversed the order of creating the subscription in DB and creating the Google subscription.